tree/dataframe/src/RLoopManager.cxx

```diff
@@ -139,7 +141,16 @@ void ExploreBranch(TTree &t, std::set<std::string> &bNamesReg, ColumnNames_t &bN
       TBranch *subBranch = static_cast<TBranch *>(sb);
       auto subBranchName = std::string(subBranch->GetName());
       auto fullName = prefix + subBranchName;
-
+      auto prefixTokens = ROOT::Split(prefix, ".", true /*skipEmpty*/);
+      auto subBranchNameTokens = ROOT::Split(subBranchName, ".", true /*skipEmpty*/);
+      if (!prefixTokens.empty() && !subBranchNameTokens.empty() && prefixTokens.back() == subBranchNameTokens.front()) {
+         fullName = std::string{};
+         for (const auto &s : prefixTokens)
+            fullName += s + ".";
+         for (decltype(subBranchNameTokens.size()) i = 1; i < subBranchNameTokens.size(); i++)
```

Comment:
If I understand this code correctly you don't really need to Split the strings; can't you instead find the last occurrence of `.` in `prefix`, the first occurrence of `.` in `subBranchName` and then, after matching the substrings, do
```
fullName = prefix + subBranchName.substr(firstDotIdx);
```
?